### ISSUE TITLE
Mistake in stretching matrix in layered model

### ISSUE THREAD

**Issue by mpudig** [pyqg/pyqg#325]
Hi,

We (in fact, @Empyreal092) noticed a possible mistake in how the reduced gravities are defined in `layered_model` (https://github.com/pyqg/pyqg/blob/master/pyqg/layered_model.py#L176).

It is our understanding that in the multi-layer QG equations, one should divide by a reference density in the reduced gravities (see [Vallis, 2017](https://empslocal.ex.ac.uk/people/staff/gv219/aofd/) p.186), rather than by the density of each layer as is currently done in `layered_model` (as well as in its pyqg documentation). 

Could anyone who was part of model development (@rabernat, @MFJansen, @cesar-rocha, @francispoulin perhaps) comment on whether this is in fact a typo, or if there was a reason behind this choice for the reduced gravities? I'd be happy to push through the changes necessary if required.

**Comment by francispoulin** [comment]
Thanks @mpudig for the question, and sorry for the late reply.

I remember going through this calculation in Vallis' textbook years ago.  The textbook uses a different value of density than what I would expect.  If you look at the following paper, I believe we use the correct value in our two-layer formulation.

https://journals.ametsoc.org/view/journals/phoc/52/11/JPO-D-21-0318.1.xml

In the context of the Boussinesq approximation, where the density is not allowed to change very much, I'm not sure if this can every make a difference. 

If you tell us what specific change you suggest I'm happy to take it into consideration.

(I realize this reply is 1.5 years old and maybe you have lost interest, but I thought I would replyl, just in case.)

**Comment by mpudig** [comment]
Hi @francispoulin 

No problem. I believe the derivation of the layered QG equations from Boussinesq stacked shallow-water will result in a constant reference density in the denominator of all reduced gravities. We commented on and pushed a similar change into the layered QG module of GeophysicalFlows.jl ([here](https://github.com/FourierFlows/GeophysicalFlows.jl/pull/343) with further discussion on the effect of _not_ using a constant reference density [here](https://github.com/FourierFlows/GeophysicalFlows.jl/discussions/325)).

Our solution was to ask for buoyancy as an input rather than density. I believe an equally fine alternative is to replace https://github.com/pyqg/pyqg/blob/4e8f50f13d9fffccd3fce53aa27d8183b7827819/pyqg/layered_model.py#L176
by something like:

`self.gpi = self.g*(self.rhoi[1:]-self.rhoi[:-1])/self.rhoi[0] # m s^-2`

as in Vallis's text (2017; p186).

**Comment by francispoulin** [comment]
I agree that is a much better way of doing things.  Thanks for explaining this so clearly.

I can make the change unless you would rather make a PR?

**Comment by mpudig** [comment]
Happy for you to make the change! As I guess either version could be implemented (buoyancy or density with correct reference density). I can't remember if there's an argument for one being more appropriate for this type of module...
